test/mv_opentelemetry/span_tracer_test.exs

```diff
@@ -0,0 +1,39 @@
+defmodule MvOpentelemetry.SpanTracerTest do
+  use MvOpentelemetry.OpenTelemetryCase
+
+  alias MvOpentelemetry.CustomSpanTracer
+
+  test "it captures the event" do
+    :otel_batch_processor.set_exporter(:otel_exporter_pid, self())
+    CustomSpanTracer.register_tracer()
+
+    :telemetry.span([:my_event, :do_stuff], %{name: "custom"}, fn ->
+      {:ok, %{result: :success}}
+    end)
+
+    assert_receive {:span, span_record}
+    assert "my_event.do_stuff" == span(span_record, :name)
+    attributes = span(span_record, :attributes)
+    assert {:"my_event.name", "custom"} in attributes
+    assert {:"my_event.result", :success} in attributes
+
+    :ok = :telemetry.detach({CustomSpanTracer, CustomSpanTracer})
+  end
+
+  test "it captures the exception" do
+    :otel_batch_processor.set_exporter(:otel_exporter_pid, self())
+    CustomSpanTracer.register_tracer()
+
+    try do
+      CustomSpanTracer.raise_test_error()
+    rescue
+      RuntimeError ->
+        assert_receive {:span, span_record}
+        assert "my_event.do_stuff" == span(span_record, :name)
+        attributes = span(span_record, :attributes)
+        assert {:error, true} in attributes
+        assert {:kind, :error} in attributes
+        :ok = :telemetry.detach({CustomSpanTracer, CustomSpanTracer})
+    end
```

Comment:
Could you use [assert_raise/2](https://hexdocs.pm/ex_unit/1.12/ExUnit.Assertions.html#assert_raise/2) instead? 🤔

Reply:
I don't think so - I'm not really interested in the error itself, but I'm looking at what happens after that. I could probably use a task to fire it off, but this way it seems more readable.